src/shared/json.c

```diff
@@ -1066,6 +1066,50 @@ double json_variant_real(JsonVariant *v) {
         return 0.0;
 }
 
+int json_variant_bytes(JsonVariant *v, uint8_t **ret, size_t *ret_size) {
+        _cleanup_free_ uint8_t *buf = NULL;
+        size_t n;
+
+        assert(ret);
+        assert(ret_size);
+
+        if (!v)
+                goto not_found;
+        if (!json_variant_is_array(v))
+                return -EINVAL;
+
+        n = json_variant_elements(v);
+        if (n == 0)
+                goto not_found;
+
+        buf = new(uint8_t, n);
```

Comment:
tbh i'd allocate one more byte here, silently, and zero initialize it. just in case. as a safety net in case a consumer ends up using this as a string or so, by accident. add a brief comment, but don#t advertise this otherwise, so that people don't rely on it. just defensive programming